mqtt/init.lua

```diff
@@ -81,6 +81,183 @@ function mqtt.run_sync(cl)
 	end
 end
 
+
+--- Validates a topic with wildcards.
+-- @param t (string) wildcard topic to validate
+-- @return topic, or false+error
```

Comment:
IMO, the validation function should return true on success, not its argument

Reply:
as mentioned above, if it returns the value, or nil+err, it can be used as input for an assertion. A validation function is less useful if you first have to validate the input values before calling the validator.